### ISSUE TITLE
Our current Select component is problematic

### ISSUE THREAD

**Issue by dougmacknz** [cultureamp/kaizen-design-system#1767]
## Issues

* Pulled up in the 2020 accessibility audit with _many_ accessibility issues and in need of a rewrite, which we did as a bespoke component in Capture.
* Doesn't function well as a form `<select>`.
  * No ability to easily add a label and no ability to link a label and the select properly. See https://cultureamp.slack.com/archives/C0189KBPM4Y/p1626401032437000
  * Not part of the `form` package and doesn't have consistent naming with our other inputs (Text**Field**, TextArea**Field**)

(could be more that I'm forgetting, please chime in if you have more to add)

## A way forward

The most likely way forward I see is:

* Create a new `SelectField` component in our `form` package. This should just be a native `<select>` under the hood for now and can be used when consumers don't need any search/autocomplete functionality.
* Create a new `Autocomplete/Combobox` component (see below)
* Deprecate our existing `Select` component.

Another option is that we look at building the autocomplete functionality into the `SelectField` component, but I have a feeling that the autocomplete functionality is complicated enough to warrant a separate component.

There's also a question of whether the new Autocomplete component should be considered part of `form` or not, but that is something that could be decided later down the line.

## Autocomplete functionality

For the autocomplete functionality, it would be great to outsource it as much as possible because it was a *huge* amount of work to create the bespoke component for Capture, and there were still some issues with it on VoiceOver/iOS.

https://react-spectrum.adobe.com/blog/building-a-combobox.html was just released and look like a very promising library for us to shim over for an accessible autocomplete component.

**Comment by dougmacknz** [comment]
Here's an example of an autocomplete in the wild (Coles online shopping) that's not a `<select>`, but an `<input type="search" />`, so basically our `SearchField` component that hasn't been built yet: 
![image](https://user-images.githubusercontent.com/1811583/125891492-9a866fcd-242f-41dd-b54a-be7ab36d4697.png)

We'll want to be able to support in this in the future, so back to the two options posed above, we could either:
* Create a single `Autocomplete` component that could either be used as a search input, or a select.
* Add the autocomplete functionality into `SearchField` and `SelectField` (ideally with common foundational code for the autocomplete stuff).

The former might make more sense code-wise, but I also worry that it would be confusing for consumers.

> You want a search field? Use the `SearchField` component.

> Done, now how do I add some suggestions?

> Oh, in that case you must use `Autocomplete`

**Comment by talbet** [comment]
Another issue I've noticed is that whenever you click inside the select component, it toggles the drawer state open/closed regardless of intent. You'll see what I mean if you open up a multi-select examples and add 3 or 4 elements and then remove them one by one. The first removal click opens the drawer, the second removal closes the drawer and so on.
